src/js/grapl-cdk/swarm/swarm_setup.py

```diff
@@ -0,0 +1,238 @@
+"""Usage: python3 swarm_setup.py $GRAPL_DEPLOY_NAME"""
+import json
+import logging
+import os
+import sys
+import time
+
+import boto3
+import botocore
+
+from typing import Any, Iterator, List, Tuple
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel("INFO")
+LOGGER.addHandler(logging.StreamHandler(stream=sys.stdout))
+
+IN_PROGRESS_STATUSES = {
+    "Pending",
+    "InProgress",
+    "Delayed",
+}
+
+
+def _swarm_instances(ec2: Any) -> Iterator[Tuple[str, str, str]]:
+    """Yields tuples of (instance_id, private_ip, hostname) for all the
+    instances in the SwarmASG.
+
+    """
+    result = ec2.describe_instances(
+        Filters=[{"Name": "tag:Name", "Values": ["Grapl/swarm/SwarmCluster/SwarmASG"]}]
+    )
+    for reservation in result["Reservations"]:
+        for instance in reservation["Instances"]:
+            if instance["State"]["Name"] != "terminated":
+                yield instance["InstanceId"], instance["PrivateIpAddress"], instance[
+                    "PrivateDnsName"
+                ]
+
+
+def _get_command_result(ssm: Any, command_id: str, instance_id: str) -> str:
+    """Poll until the command result is available for the given
+    command_id. Returns the command result.
+
+    """
+    while 1:
+        commands = ssm.list_commands(CommandId=command_id)
+        if commands["Commands"][0]["Status"] in IN_PROGRESS_STATUSES:
+            time.sleep(2)
+        else:
+            break
```

Comment:
could we do either a print("Checking status...") or an eventual-timeout here?

i find that "while true: silently try something, then sleep, forever and ever" are hard to debug for consumers, in case something's genuinely fucked up

it's the root of a thing that screwed up andrea the other day
https://github.com/grapl-security/grapl/pull/429/files#diff-07bf86199f1865f81c28ab7cdb4d0b3572d6b36587479b3a617b6ce066fe01beL53-L69

Reply:
done